code/game/objects/items/storage/boxes/job_boxes.dm

```diff
@@ -190,9 +190,12 @@
 		return ITEM_INTERACT_BLOCKING
 	qdel(tool)
 	loc.balloon_alert(user, "wheels added, honk!")
-	var/obj/item/bot_assembly/honkbot/A = new
-	qdel(src)
-	user.put_in_hands(A)
+	var/obj/item/bot_assembly/honkbot/assembly = new(drop_location())
+	if (user.is_holding(src))
+		qdel(src)
+		user.put_in_hands(assembly)
+	else
+		qdel(src)
```

Comment:
I just don't really like how this pattern of the if-else differing one line looks
```dm
	if(user.is_holding(src))
		qdel(src)
		user.put_in_hands(assembly)
	else
		qdel(src)
```
the alternatives don't feel that much cleaner though so, eh